cfgov/v1/models/base.py

```diff
@@ -225,6 +225,43 @@ def get_next_appropriate_siblings(self, hostname, inclusive=False):
     def get_prev_appropriate_siblings(self, hostname, inclusive=False):
         return self.get_appropriate_siblings(hostname=hostname, inclusive=inclusive).filter(path__lte=self.path).order_by('-path')
 
+    def get_context(self, request, *args, **kwargs):
+        context = super(CFGOVPage, self).get_context(request, *args, **kwargs)
+        for hook in hooks.get_hooks('cfgovpage_context_handlers'):
+            hook(self, request, context, *args, **kwargs)
+        return context
+
+    def serve(self, request, *args, **kwargs):
+        """
+        If request is ajax, then return the ajax request handler response, else
+        return the super.
+        """
+        if request.is_ajax():
+            return self.serve_ajax(request)
+
+        return super(CFGOVPage, self).serve(request, *args, **kwargs)
+
+    def serve_ajax(self, request):
+        """
+        Attempts to retreive form_id from the POST request and returns a JSON
+        response.
+
+        If form_id is found, it returns the response from the block method
+        retrieval.
+
+        If form_id is not found it returns an error response.
+        """
```

Comment:
I think I understand what this is trying to do, although this seems to introduce certain hidden assumptions, namely that Wagtail page submissions via AJAX will always include a `form_id` parameter. This means in future we couldn't support other arbitrary AJAX submissions.

Also, should this return an `HTTP 400 Bad Request` instead of 200 to be consistent with RESTful status codes?